doc: Documentation on contributors (#280)
CanalTP/kirin: readme.md

```

#### Contributors

##### GET
Returns a list of contributors present in the database

```
curl -X GET 'http://localhost:5000/contributors'
```
``` json
{
  "contributors": [
    {
      "feed_url": "feed_url_not_used_for_cots",
      "navitia_coverage": "sncf",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4hhhhhhhhhhhhhhhh",
      "connector_type": "cots",
      "id": "realtime.cots"
    },
    {
      "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
      "navitia_coverage": "ca-qc-sherbrooke",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
      "connector_type": "gtfs-rt",
      "id": "realtime.sherbrooke"
    }
  ]
}
```
Returns a specific contributor present in the database
```
curl -X GET 'http://localhost:5000/contributors/realtime.sherbrooke'
```

``` json
{
  "contributors": [
    {
      "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
      "navitia_coverage": "ca-qc-sherbrooke",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
      "connector_type": "gtfs-rt",
      "id": "realtime.sherbrooke"
    }
  ]
}
```

##### POST

> To create a new contributor, parameters need to be given in the query's string or in json send in the body

Reviewer: Are you sure about the params in url? Anyway I wouldn't mention that.
```suggestion
To create a new contributor, parameters need to be given in json sent in the body
```

###### post Parameters
Property | Type | Description
--- | --- | ---
id | String, Required | Unique code of the contributor
navitia_coverage | String, Required | Navitia coverage to be used while calling navitia to retrieve corresponding element
navitia_token | String, Optional | Navitia token to be used while calling navitia to retrieve corresponding element
feed_url | String, Optional | Url to retrieve the realtime information feed (for polled sources)
connector_type | Enum, Required | Type of connector (possible values are `cots`, `gtfs-rt`)

> is_active | Boolean, Required | Used to activate/deactivate the kirin service for the contributor

Reviewer: This one is not required at POST, no?

> is_active | Boolean, Required | Used to activate/deactivate the kirin service for the contributor

Reviewer 2: Why not a link to " documentation/internal_format.md" ?

```
curl -X POST 'http://localhost:5000/contributors/' -d'{"feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb", "navitia_coverage": "ca-qc-sherbrooke", "is_active": true, "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a", "connector_type": "gtfs-rt", "id": "realtime.sherbrooke"}' -H'content-type: application/json'
```
``` json
{
  "contributor": {
    "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
    "navitia_coverage": "ca-qc-sherbrooke",
    "is_active": true,
    "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
    "connector_type": "gtfs-rt",
    "id": "realtime.sherbrooke"
  }
}
```

##### PUT

> Modifies a contributor and has same parameters as POST

Reviewer: Force newline:
```suggestion
Modifies a contributor and has same parameters as POST.\
```

> When a parameter is missing, it is not changed.

Reviewer: ```suggestion
When a parameter is missing, it is not changed.\
```

> When PUTing the exact result of a GET (using disable_geojson or not), nothing is changed.

Reviewer: ```suggestion
When PUTing the exact result of a GET, nothing is changed.
```

Reviewer 2: Do we really need this sentence? Seems obvious

```
curl -X PUT 'http://localhost:5000/contributors/' -d'{"feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb", "navitia_coverage": "ca-qc-sherbrooke", "is_active": true, "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a", "connector_type": "gtfs-rt", "id": "realtime.sherbrooke"}' -H'content-type: application/json'
```
``` json
{
  "contributor": {
    "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
    "navitia_coverage": "ca-qc-sherbrooke",
    "is_active": true,
    "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
    "connector_type": "gtfs-rt",
    "id": "realtime.sherbrooke"
  }
}
```

##### DELETE
Deletes a specific contributor

```
curl -X DELETE 'http://localhost:5000/contributors/realtime.toto'
```

> ##### Status (GET)

Reviewer: To match the contributor section
```suggestion
#### Status (GET)
```

```

``` json
{
  "contributors": [
    {
      "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
      "navitia_coverage": "ca-qc-sherbrooke",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
      "connector_type": "gtfs-rt",
      "id": "realtime.sherbrooke"
    }
  ]
}
```

##### POST
To create a new contributor, parameters need to be given in json sent in the body as described [here](documentation/internal_format.md)

```
curl -X POST 'http://localhost:5000/contributors/' -d'{"feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb", "navitia_coverage": "ca-qc-sherbrooke", "is_active": true, "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a", "connector_type": "gtfs-rt", "id": "realtime.sherbrooke"}' -H'content-type: application/json'
```
``` json
{
  "contributor": {
    "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
    "navitia_coverage": "ca-qc-sherbrooke",
    "is_active": true,
    "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
    "connector_type": "gtfs-rt",
    "id": "realtime.sherbrooke"
  }
}
```

##### PUT
Modifies a contributor and has same parameters as POST.

> When a parameter is missing, it is not changed.

Reviewer: Did you test it? I think we reject, no?
As working with partial input would be a PATCH.

Reviewer 2: I tested it. It is the current behavior. PUT with only one parameter (beside `id`) changes only the parameter and keeps the others

curl 'http://localhost:5000/'
```

> ##### Contributors (add, update or delete)

Reviewer: ```suggestion
##### Contributors
```
Usual vocable is more "create" than "add" (for CRUD)
But I would just move that to subsections titles.

> Returns a list of contributors present in the database

Reviewer: This section is just the "read" subsection of "Contributors" section.
My opinion is this should be a separate subsection and 3 more subsections for create, update and delete should be added.

```
curl 'http://localhost:5000/contributors'
```
``` json
{
  "contributors": [
    {
      "feed_url": "feed_url_not_used_for_cots",
      "navitia_coverage": "sncf",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4hhhhhhhhhhhhhhhh",
      "connector_type": "cots",
      "id": "realtime.cots"
    },
    {
      "feed_url": "http://0.0.0.0./civilia/TTT/pb/tripUpdates.pb",
      "navitia_coverage": "ca-qc-sherbrooke",
      "is_active": true,
      "navitia_token": "9489dd5f-46b4-mmmmmmmmmm3bfba0c71e8a",
      "connector_type": "gtfs-rt",
      "id": "realtime.sherbrooke"
    }
  ]
}
```

> Returns the contributor present in the database

Reviewer: minor
```suggestion
Returns a specific contributor present in the database
```

for both Kirin and Kraken (the navitia core calculator).

- In Kirin:

>     - Add a contributor 'cots' using the end point /contributors

Reviewer 2: ```suggestion
    - Add a contributor 'cots' using the end point `/contributors`
```
